### DISCUSSION TITLE
Dúvida sobre top level statements em C#

### DISCUSSION THREAD

**Discussion by Lorenzogrocha** [balta-io/forum#614] [Tira Dúvidas] [answered]
Olá. A minha dúvida é a seguinte: Quando usar e quando não usar top level statements ? No mercado de trabalho seria como ?
Me refiro aos namespace, using, classe main...

Exemplo **sem** top level statements :

`namespace ScreenSoundTwo
{
    internal class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello, World!");
        }
    }
}`

Seria o mesmo que isto **com uso** do top level statements :

`Console.WriteLine("Hello, World!");`

Eu particularmente prefiro da primeira forma, mas vi que atualmente o dotnet 8 segue nesse outro modelo. Sou iniciante, talvez seja uma dúvida meio besta.
Usando este recurso, eu acho que me perco um pouco sobre o que seria executado primeiro e na organização, acabo preferindo o uso da Main.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Lorenzogrocha, como vai?

_**Olá. A minha dúvida é a seguinte: Quando usar e quando não usar top level statements ?**_ 
- A gosto do programador... eu prefiro usar o top-level, acho mais simples, mas não vejo problemas em não usar

**_No mercado de trabalho seria como?_**
- O padrão a partir do .NET 6 é usando o top-level, então depende do projeto que você for trabalhar...

**_Me refiro aos namespace, using, classe main..._**
- Mesma coisa aqui... mesma resposta do item anterior

Em resumo: Saiba os dois, use o qual você se sente melhor... No mercado, nos projetos, siga sempre o padrão do projeto!!!

**Comment by Lorenzogrocha** [comment]
Obrigado!
